### PR TITLE
Update folly

### LINKED ISSUE #6138: Building Solution: The contents of <bit> are available only with C++20 or later....
This started showing up frequently in build logs. This is pretty disruptive to build logs in `run-windows` which flash between this error and the file.

Likely related to the Folly upgrade.

### PR DESCRIPTION
**Pull request by asklar**
Fixes #6138 

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native-windows/pull/6187)

### CODE REVIEW

File: vnext/Directory.Build.props [hunk 1]
```diff
@@ -17,7 +17,7 @@
       The PR (windows-vs-pr.yml) and CI (publish.yml() turn it back on.
     -->
     <EnableSourceLink Condition="'$(EnableSourceLink)' == ''">false</EnableSourceLink>
-    <FollyVersion>2020.09.14.00</FollyVersion>
+    <FollyVersion>2020.10.05.00</FollyVersion>
```
Comment (dannyvv) [inline vnext/Directory.Build.props:20]: Woudl you mind chcking the failure in the publish step? It has been failing to copy the files in Layout-Headers.ps1 to inc\folly which makes packing up nuget files fail. I see you paraemeterized it there to read this file, but when looking in the build artifact created, the files are not there...

**Review by dannyvv — APPROVED** [review]

**Comment by asklar** [comment]
@dannyvv I'm not getting any pipelines... any idea what's going on?

**Comment by NickGerleman** [comment]
New PRs aren't making there way from GitHub to AZP at all https://dev.azure.com/ms/react-native-windows/_build?definitionId=210&_a=summary

**Pull request closed without merging**